ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandRunnerTest.java

```diff
@@ -286,9 +300,32 @@ public void shouldProcessPartialListOfCommandsOnIncomaptibleCommandInFetch() {
     inOrder.verify(statementExecutor).handleRestore(eq(queuedCommand2));
 
     assertThat(commandRunner.checkCommandRunnerStatus(), is(CommandRunner.CommandRunnerStatus.DEGRADED));
+    assertThat(commandRunner.getCommandRunnerDegradedWarning(), is(INCOMPATIBLE_COMMANDS_ERROR_MESSAGE));
+    assertThat(commandRunner.getCommandRunnerDegradedReason(), is(CommandRunner.CommandRunnerDegradedReason.INCOMPATIBLE_COMMAND));
     verify(statementExecutor, never()).handleRestore(queuedCommand3);
   }
 
```

Comment:
Can we add test cases that ensure that within a batch, we process everything that is valid (so all commands for which the version is valid, and all commands up to the backup corruption point)?

Reply:
We're implicitly assuming that commands returned from the CommandStore aren't corrupted.

I also don't think we're able to have fine tuned corruption detection in CommandRunner. Assuming we did have corrupted commands returned from the CommandStore in the batch, the function to check for corruption doesn't indicate which command is corrupted so all we'd know is that corruption was detected after the batch was fetched.

There should be existing tests testing when the incompatibleCommandChecker throws an exception while checking a batch and everything up to the incompatible command is processed by the InteractiveStatementExecutor